modules/data.remote/R/extract_phenology_MODIS.R

```diff
@@ -0,0 +1,93 @@
+                                                package_method = "MODISTools", 
+                                                QC_filter = FALSE, 
+                                                progress = TRUE)
+
+    qa_data <- PEcAn.data.remote::call_MODIS(outdir = NULL, 
+                                           var = "QA", 
+                                           site_info = site_info, 
+                                           product_dates = c(start_YEARDOY, end_YEARDOY),
+                                           run_parallel = as.logical(run_parallel), 
+                                           ncores = ncores, 
+                                           product = "MCD12Q2", 
+                                           band = "QA_Detailed.Num_Modes_01",
+                                           package_method = "MODISTools", 
+                                           QC_filter = FALSE, 
+                                           progress = TRUE)
+
+    #Example function for unpacking QA for each date phenometric from MCD12Q2_Collection6_UserGuide
+    #in the order: Greenup, MidGreenup, Maturity, Peak, Senescence, MidGreendown, Dormancy                                        
+    UnpackDetailedQA <- function(x){
+       bits <- as.integer(intToBits(x))
+       quals <- sapply(seq(1, 16, by=2), function(i) sum(bits[i:(i+1)] * 2^c(0, 1)))[1:7]
+       return(quals)
+      }
+    #Extract QA for leaf-on and leaf-off dates. Values are in the range 0–3 corresponding to “best”, “good”, “fair”, and “poor”.
+    qa_ind<-matrix(NA,nrow=length(qa_data[,1]),ncol=2)
+    for (i in seq_along(qa_data$data)){
+      qa_ind[i,]<-cbind(UnpackDetailedQA(qa_data$data[i])[2],UnpackDetailedQA(qa_data$data[i])[6])
+    }
+  
+    leafphdata <- cbind(leafon_data %>% as.data.frame %>% dplyr::select("calendar_date", "site_id", "lat", "lon", "data"), leafoff_data$data,qa_ind)%>% 
+    `colnames<-`(c("year", "site_id", "lat", "lon", "leafonday","leafoffday","leafon_qa","leafoff_qa"))
+    leafphdata$leafonday[leafphdata$leafonday==32767]<-NA #exclude the data with fill values
+    leafphdata$leafoffday[leafphdata$leafoffday==32767]<-NA
+    leafphdata$leafonday[leafphdata$leafon_qa==3]<-NA #exclude the data when QA is poor
+    leafphdata$leafoffday[leafphdata$leafoff_qa==3]<-NA
+    leafphdata$leafonday[leafphdata$leafonday>leafphdata$leafoffday]<-NA #exclude the data when leaf-on date is larger than leaf-off date
+    leafphdata$leafonday<-lubridate::yday(as.Date(leafphdata$leafonday)) #convert the dates to Day-of-Year format
+    leafphdata$leafoffday<-lubridate::yday(as.Date(leafphdata$leafoffday))
+    leafphdata$year<-lubridate::year(leafphdata$year)
+    leafphdata$site_id<-as.character(leafphdata$site_id)
+    
+    PEcAn.logger::logger.info(paste0("Storing results in: ",file.path(outdir,"leaf_phenology.csv")))
+    utils::write.csv(leafphdata, file = file.path(outdir, "leaf_phenology.csv"), row.names = FALSE)
+    return(leafphdata)
```

Comment:
function appears to return the phenology data itself, but code in do_conversions assumes that the function is returning the PATH to the phenology data. I suspect the latter is correct given that write.configs is looking for a file path rather than downloading the data in real time.

Reply:
```suggestion
    return( file.path(outdir, "leaf_phenology.csv"))
```